docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+
+We have implemented backend role-based access in which members of the same back-end roles will be able to access each other’s resources. This is similar to alerting and anomaly detection plugin [backend role based access](https://opensearch.org/docs/latest/observing-your-data/alerting/security/#advanced-limit-access-by-backend-role). Additionally we also let users define an ml resource as public or private. This advanced security access is available only on a cluster with [Security plugin](https://opensearch.org/docs/latest/security/index/).
+
+Note: 
+- Model access control is experimental feature. If you see any bug or have any suggestion, feel free to cut Github issue.
+
+
+
+## Setup
+
+For Backend Role based security, it is not enabled by default. To enable/disable it, customers can toggle elasticsearch cluster setting shown below. If this setting is not enabled, a resource can be accessed by anyone.
+
+```
+PUT /_cluster/settings
+{
+  "persistent" : {
+    "plugins.ml_commons.model_access_control_enabled" : false 
+  }
+}
+```
+
+For the purpose of this step-by-step tutorial, let us consider admin and four other users with "ml_full_access" permission role i.e. all the users have permissions to perform all actions on ml resources. For more information on users and roles, refer to https://opensearch.org/docs/latest/security/access-control/users-roles/
+
+| Users   | Backend role                      |
+|:--------|:----------------------------------|
+| `User1` | IT, HR                            |
+| `User2` | IT                                |
+| `User3` | Finance                           |
+| `User4` | -                                 |
+| `Admin` | Admin can assign any backend role |
+
+
+
+## Model groups
+
+Users can register different versions to the same model. For ease of control, these versions can be registered under a same model group. So, user first needs to create a model group and then start registering model versions to it. A model group can have any number of versions associated with it.
+
+The access control in ml-commons is implemented at the _model group level_ so that all versions of the same model share the same access mode as that of their model group. That is, if user has access to one version, then he/she should have access to all versions of the model group.
+
+When creating a model group, user should also specify the access mode to it. Depending on the access mode, other users will be able to access the versions of a model group or create new versions to it.
+
+If a user tries to register/deploy/undeploy/predict/delete/search/get model versions, first we check if the user has access to the model. Otherwise, we throw exception saying user does not have access to the model specified.
+
+A user is considered a model _owner_ when he/she creates a new model group and register its first version. When a model owner creates a model group, the owner can specify one of the following _access modes_ for this model group:
```

Comment:
> A user is considered a model _owner_ when he/she creates a new model group and register its first version.

This sentence says user need to create new group and register first version. What if user register model version directly to a model group? Will that user be model owner of the model version ?

Reply:
If I understand correctly model owner is something related with model group. If a user register a model group , the user will be the owner of the model group.

Let's user "model group" and "model version" to avoid confusion?